### DISCUSSION TITLE
Issue with auth cookie

### DISCUSSION THREAD

**Discussion by erwanevn** [adonisjs/core#4289] [Help]
Hi!

I'm having a problem when I try to use cookies on front-end side. I'm using NextJS for front and when i try to send a request to the server using cookies, all works but when i try the same thing on middlewares, the request doesnt includes cookies. I thing the issue come from back-end, sorry if it's front
I leave you the two projects if you want to recreate it, thanks !! ;)

Front: https://github.com/erwan-eveno/ulysse-client.git
Back: https://github.com/erwan-eveno/ulysse-server.git

**Comment by tomgobich** [comment]
First, you may want to register the `SilentAuth` middleware within your application, inside `start/kernel.ts`. This will automatically populate the `auth.user` if a user is logged in. Otherwise, you'll need to manually call `auth.check()` on the requests you need your authenticated user.
```ts
Server.middleware.register([
  () => import("@ioc:Adonis/Core/BodyParser"),
  () => import('App/Middleware/SilentAuth') // <--
]);
```

Second, I'm not sure whether you're running into this it looks like all your calls are currently client-side in your Next app, but I'll mention it just in case. When you're working with NextJS/NuxtJS along with an AdonisJS API note that you're dealing with two servers, your Next/Nuxt server and your AdonisJS server.

When you send a request from the server side of your Next/Nuxt server to your AdonisJS server, your cookies won't automatically be attached to the request like they are for client-side requests.

To remedy this you'll need to:
1. Pluck the cookies off the request sent from the browser to your Next/Nuxt server
2. Attach the cookies to the request sent from your Next/Nuxt server to your AdonisJS API

I'm not sure what this looks like on NextJS, it's been years since I've touched it, but I have a lesson that walks through this for Nuxt here, which may be helpful
https://adocasts.com/series/adonisjs-in-15/lessons/adonisjs-api-nuxt-3-authentication-in-15-minutes
